src/linear_algebra/basic.lean

```diff
@@ -569,14 +595,19 @@ section semiring
 variables [semiring R] [add_comm_monoid M] [module R M]
 
 instance endomorphism_semiring : semiring (M →ₗ[R] M) :=
-by refine_struct
-  { mul := (*),
-    one := (1 : M →ₗ[R] M),
-    zero := 0,
-    add := (+),
-    npow := @npow_rec _ ⟨1⟩ ⟨(*)⟩,
-    .. linear_map.add_comm_monoid, .. };
-intros; try { refl }; apply linear_map.ext; simp {proj := ff}
+{ mul := (*),
+  one := (1 : M →ₗ[R] M),
+  zero := 0,
+  add := (+),
+  npow := @npow_rec _ ⟨1⟩ ⟨(*)⟩,
+  left_distrib := λ f g h, by { ext, simp },
+  right_distrib := λ f g h, by { ext, simp },
+  zero_mul := λ f, by { ext, simp },
+  mul_zero := λ f, by { ext, simp },
+  mul_assoc := λ f g h, by { ext, simp [mul_assoc] },
+  one_mul := λ f, by { ext, simp },
+  mul_one := λ f, by { ext, simp },
+  .. linear_map.add_comm_monoid }
```

Comment:
#9336 has a shorter proof of this, I think we should merge it first

Reply:
Fair enough.